src/ispyb/cli/job.py

```diff
@@ -0,0 +1,471 @@
+        sweeps.append(values)
+
+    if options.dcid:
+        match = re.match(r"^([0-9]+)$", options.dcid)
+        if not match:
+            sys.exit("Invalid data collection id: " + options.dcid)
+        dcid = int(options.dcid)
+    else:
+        dcid = None
+
+    if not sweeps:
+        if not dcid:
+            sys.exit(
+                "When creating a processing job you must specify at least one data collection sweep or a DCID"
+            )
+
+        dc_info = i.get_data_collection(dcid)
+        start = dc_info.image_start_number
+        number = dc_info.image_count
+        if not start or not number:
+            sys.exit("Can not automatically infer data collection sweep for this DCID")
+        end = start + number - 1
+        sweeps = [(dcid, start, end)]
+        print(f"Using images {start} to {end} for data collection sweep")
+
+    parameters = []
+    for p in options.parameters:
+        if ":" not in p:
+            sys.exit("Invalid parameter specification: " + p)
+        parameters.append(p.split(":", 1))
+
+    if zocalo:
+        trigger_variables = []
+        for p in options.triggervariables:
+            if ":" not in p:
+                sys.exit("Invalid trigger variable specification: " + p)
+            trigger_variables.append(p.split(":", 1))
+
+    jp = i.mx_processing.get_job_params()
+    # _job_params = StrictOrderedDict([('id', None), ('datacollectionid', None), ('display_name', None), ('comments', None), ('recipe', None), ('automatic', None)])
+    jp["automatic"] = options.source == "automatic"
+    jp["comments"] = options.comment
+    jp["datacollectionid"] = dcid or sweeps[0][0]
+    jp["display_name"] = options.display
+    jp["recipe"] = options.recipe
+    print("Creating database entries...")
+
+    jobid = i.mx_processing.upsert_job(list(jp.values()))
+    print(f"  JobID={jobid}")
+    for key, value in parameters:
+        jpp = i.mx_processing.get_job_parameter_params()
+        # _job_parameter_params = StrictOrderedDict([('id', None), ('job_id', None), ('parameter_key', None), ('parameter_value', None)])
+        jpp["job_id"] = jobid
+        jpp["parameter_key"] = key
+        jpp["parameter_value"] = value
+        jppid = i.mx_processing.upsert_job_parameter(list(jpp.values()))
+        print(f"  JPP={jppid}")
+
+    for sweep in sweeps:
+        jisp = i.mx_processing.get_job_image_sweep_params()
+        # _job_image_sweep_params = StrictOrderedDict([('id', None), ('job_id', None), ('datacollectionid', None), ('start_image', None), ('end_image', None)])
+        jisp["job_id"] = jobid
+        jisp["datacollectionid"] = sweep[0]
+        jisp["start_image"] = sweep[1]
+        jisp["end_image"] = sweep[2]
+        jispid = i.mx_processing.upsert_job_image_sweep(list(jisp.values()))
+        print(f"  JISP={jispid}")
+
+    print(f"All done. Processing job {jobid} created")
+    print()
+    if zocalo:
+        if options.trigger:
```

Comment:
Could this be `
```suggestion
    if zocalo and options.trigger:
```
and remove one level of indentation? It seems no harm in printing out the `zocalo.go` command even if zocalo is not available.

We're also using the presence of the `zocalo` module in the current python environment as a proxy for the presence of the `zocalo.go` command line program - is it possible that someone could have zocalo available in a _different_ Python environment?

Reply:
I see now that the `--trigger` option is only set if `zocalo` is available, so perhaps the nested indentation does make sense after all...

Reply:
there is also an `else` clause for the inner `if`